owncloud/gallery
# Official drag and drop support

Solution for #405, replaces #519 
- You can drag images and albums to any album or the breadcrumb
- If successful, the photowall is redrawn
- If you're not allowed to make the change, elements go back to their initial position
### Possible improvements
- ~~Drop targets could be nicer~~ Done
- ~~Dragged albums maybe need a border?~~ After extensive use, I don't think so
### Screencasts

http://screencast-o-matic.com/watch/cDnl0rhWt7

**Screenshot**
![dropzones](https://cloud.githubusercontent.com/assets/323220/13683291/357dd19c-e706-11e5-9435-0586b399c78f.png)
### Test plan

Name at least one file and one folder with numbers only

As a logged-in user:
- Drag a file to a folder and drop it
- Drag a file to a breadcrumb and drop it
- Drag a folder to another folder and drop it
- Drag a folder to a breadcrumb and drop it
- Make sure sometimes the number folder is the target and sometimes the dragged object
- **Test on as many browsers as you can. Some browsers don't handle authorization the same way**

Do the same on the public side, but make sure you've added the "editing" permission to the link.
### Caveats

Read the FAQ in #405 
### TODO
- [x] Create our own instance of Files.Client so that we can pick the URL to use

---

@jancborchardt @jospoortvliet @setnes @demattin @arkascha @elpraga @sualko @bugsbane @mmattel
@karlitschek @LukasReschke @tahaalibra @viraj96 @imjalpreet @mbtamuli @rahulgoyal030
